app/ts/components/SVGBlockie.tsx

```diff
@@ -0,0 +1,97 @@
+// ported directly from https://github.com/DarkFlorist/TheInterceptor
+
+function generateIdenticon(options: { seed: string; size?: number }) {
+	// NOTE -- Majority of this code is referenced from: https://github.com/alexvandesande/blockies
+	// Mostly to ensure congruence to Ethereum Mist's Identicons
+
+	// The random number is a js implementation of the Xorshift PRNG
+	const randseed = new Array(4) // Xorshift: [x, y, z, w] 32 bit values
+
+	function seedrand(seed: string) {
+		for (let i = 0; i < randseed.length; i++) {
+			randseed[i] = 0
+		}
+		for (let i = 0; i < seed.length; i++) {
+			randseed[i % 4] = (randseed[i % 4] << 5) - randseed[i % 4] + seed.charCodeAt(i)
+		}
+	}
+
+	function rand() {
+		// based on Java's String.hashCode(), expanded to 4 32bit values
+		const t = randseed[0] ^ (randseed[0] << 11)
+
+		randseed[0] = randseed[1]
+		randseed[1] = randseed[2]
+		randseed[2] = randseed[3]
+		randseed[3] = randseed[3] ^ (randseed[3] >> 19) ^ t ^ (t >> 8)
+
+		return (randseed[3] >>> 0) / ((1 << 31) >>> 0)
+	}
+
+	function createColor() {
+		// saturation is the whole color spectrum
+		const h = Math.floor(rand() * 360)
+		// saturation goes from 40 to 100, it avoids greyish colors
+		const s = rand() * 60 + 40 + '%'
+		// lightness can be anything from 0 to 100, but probabilities are a bell curve around 50%
+		const l = (rand() + rand() + rand() + rand()) * 25 + '%'
+
+		const color = 'hsl(' + h + ',' + s + ',' + l + ')'
+		return color
+	}
+
+	function createImageData(size: number) {
+		const width = size // Only support square icons for now
+		const height = size
+
+		const dataWidth = Math.ceil(width / 2)
+		const mirrorWidth = width - dataWidth
+
+		const data = []
+		for (let y = 0; y < height; y++) {
+			let row = []
+			for (let x = 0; x < dataWidth; x++) {
+				// this makes foreground and background color to have a 43% (1/2.3) probability
+				// spot color has 13% chance
+				row[x] = Math.floor(rand() * 2.3)
+			}
+			const r = row.slice(0, mirrorWidth)
+			r.reverse()
+			row = row.concat(r)
+
+			for (let i = 0; i < row.length; i++) {
+				data.push(row[i])
+			}
+		}
+
+		return data
+	}
+
+	const seed = options.seed.toLocaleLowerCase()
+
+	seedrand(seed)
+
+	const color = createColor()
+	const bgcolor = createColor()
+	const spotcolor = createColor()
+	const imageData = createImageData(options.size || 8)
+
+	return { imageData, color, bgcolor, spotcolor }
+}
+
+// SVGBlockie component is resizable through CSS font size
+export default function SVGBlockie({ address }: { address: string }) {
+	const pixelDensity = 8
+	const { imageData, color, spotcolor, bgcolor } = generateIdenticon({ seed: address, size: pixelDensity })
```

Comment:
Could memorize the result of generateIdenticon

Reply:
I'm not sure it is worth bothering here given how cheap this generation is (basically a loop over an 8x8 with no cryptography involved).  However, I think this code would provide memoization if that is desired.
```suggestion
	const { imageData, color, spotcolor, bgcolor } = useMemo(() => generateIdenticon({ seed: address, size: pixelDensity }), [address])
```